src/main/java/seedu/address/model/person/NameAndSchoolContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,32 @@
+package seedu.address.model.person;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+
+/**
+ * Tests that a {@code Person}'s {@code Name} matches any of the keywords given.
+ */
+public class NameAndSchoolContainsKeywordsPredicate implements Predicate<Person> {
+    private final List<String> keywords;
+
+    public NameAndSchoolContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Person person) {
```

Comment:
would this cause a problem where name will be searched when user searched for a student's school